labs/src/main/scala/org/scalalabs/basic/lab03/RecursionPatternMatchingExercise.scala

```diff
@@ -77,7 +92,8 @@ object RecursionPatternMatchingExercise {
    * List(List(1), List('A, 'B, 'C), List('a, 'b)) -> List(List(1, 'A, 'a))
    */
   def zipMultipleWithDifferentSize(in: List[List[_]]): List[List[_]] = {
-    error("fix me")
+    if (in.isEmpty||in.contains(Nil)) Nil
+    else in.map(_.head)::zipMultipleWithDifferentSize(in.map(_.tail))
   }
```

Comment:
Technically, there wasn't any pattern matching in these.... but there clearly didn't need to be, so no problem.